### DISCUSSION TITLE
Link releases with commit

### DISCUSSION THREAD

**Discussion by mattgolding** [community/community#24362] [Apps, API and Webhooks]
<p>Hi I am trying to find all the merge commits that were done on our master branch.  I am correlating this to our release tags that we generate.  I cannot seem to find the link between the release object and the commit object.  I am using the following apis to get the two data types</p>
<p>api/v3/repos/<span class="math">{repo}/releases
api/v3/repos/</span>{repo}/commits</p>
<p>Any suggestions on what I am doing wrong?  If I go to the UI and click on the release tag it shows the commit sha (short version) on the UI so I assume it is possibly to link the two.</p>

**Comment by lee-dohm** [comment, marked as answer]
<p>Hello <a class="mention" href="/mattgolding">@mattgolding</a> and welcome to the community.</p>
<p>GitHub Releases are always tied to <a href="https://git-scm.com/book/en/v2/Git-Basics-Tagging">git tags</a> rather than specific commits, even though tags themselves refer to specific commits. So, in order to find the commit that a release refers to, you have to first find the tag the release is linked to and then find the commit the tag is linked to. For example:</p>
<pre><code class="lang-bash">$ http https://api.github.com/repos/lee-dohm/last-assigned/releases/latest

----- SNIP ----
    "created_at": "2020-10-02T20:50:18Z",
    "draft": false,
    "html_url": "https://github.com/lee-dohm/last-assigned/releases/tag/v1.0.2",
    "id": 32111440,
    "name": "v1.0.2",
    "node_id": "MDc6UmVsZWFzZTMyMTExNDQw",
    "prerelease": false,
    "published_at": "2020-10-02T20:49:51Z",
    "tag_name": "v1.0.2",
----- SNIP -----
</code></pre>
<p>Notice that the <code>tag_name</code> is <code>v1.0.2</code>, then we can issue:</p>
<pre><code class="lang-bash">$ http "https://api.github.com/repos/lee-dohm/last-assigned/git/ref/tags/v1.0.2"

{
    "node_id": "MDM6UmVmMjI1OTYzODk2OnJlZnMvdGFncy92MS4wLjI=",
    "object": {
        "sha": "de87217698e438ed47d05d089c0fb4893f765872",
        "type": "commit",
        "url": "https://api.github.com/repos/lee-dohm/last-assigned/git/commits/de87217698e438ed47d05d089c0fb4893f765872"
    },
    "ref": "refs/tags/v1.0.2",
    "url": "https://api.github.com/repos/lee-dohm/last-assigned/git/refs/tags/v1.0.2"
}
</code></pre>
<p>And <code>object.sha</code> is the value that you’re looking for.</p>
<p>I hope that helps!</p>

**Comment by mattgolding** [comment]
<p>Thanks <a class="mention" href="/lee-dohm">@lee-dohm</a> exactly what I was after.</p>
<p>One more on the same topic.  Is there a way to get the commits that happened between 2 release tags ? Or would I have to get the commit against each release as you showed above and then you those 2 commit sha’s to see whats in between those ?</p>

**Comment by lee-dohm** [comment]
<aside class="quote no-group" data-username="mattgolding" data-post="3" data-topic="140267">
<div class="title">
<div class="quote-controls"></div>
 mattgolding:</div>
<blockquote>
<p>Is there a way to get the commits that happened between 2 release tags?</p>
</blockquote>
</aside>
<p>Yes, you can by using <a href="https://docs.github.com/en/free-pro-team@latest/rest/reference/repos#compare-two-commits">the compare API call</a>. For example:</p>
<pre><code class="lang-auto">http "https://api.github.com/repos/atom/atom/compare/v1.52.0...v1.51.0"
</code></pre>
<p>I hope that helps!</p>

**Comment by mattgolding** [comment]
<p>Thanks <a class="mention" href="/lee-dohm">@lee-dohm</a> will give that a try. I thought that only compared between two commit ids but your example is using the release tags so will give that a try.</p>
<p>Thanks</p>
